sw/device/lib/dif/dif_entropy.h

```diff
@@ -0,0 +1,678 @@
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_acknowledge(const dif_entropy_t *entropy,
+                                                 dif_entropy_irq_t irq);
+
+/**
+ * Checks whether a particular interrupt is currently enabled or disabled.
+ *
+ * @param entropy An entropy source handle.
+ * @param irq An interrupt type.
+ * @param[out] state Out-param toggle state of the interrupt.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_get_enabled(const dif_entropy_t *entropy,
+                                                 dif_entropy_irq_t irq,
+                                                 dif_entropy_toggle_t *state);
+
+/**
+ * Sets whether a particular interrupt is currently enabled or disabled.
+ *
+ * @param entropy An entropy source handle.
+ * @param irq An interrupt type.
+ * @param state The new toggle state for the interrupt.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_set_enabled(const dif_entropy_t *entropy,
+                                                 dif_entropy_irq_t irq,
+                                                 dif_entropy_toggle_t state);
+
+/**
+ * Forces a particular interrupt, causing it to be serviced as if hardware had
+ * asserted it.
+ *
+ * @param entropy An entropy source handle.
+ * @param irq An interrupt type.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_force(const dif_entropy_t *entropy,
+                                           dif_entropy_irq_t irq);
+
+/**
+ * Disables all interrupts, optionally snapshotting all toggle state for later
+ * restoration.
+ *
+ * @param entropy An entropy source handle.
+ * @param[out] snapshot Out-param for the snapshot; may be `NULL`.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_disable_all(
+    const dif_entropy_t *entropy, dif_entropy_irq_snapshot_t *snapshot);
+
+/**
+ * Restores interrupts from the given snapshot.
+ *
+ * This function can be used with `dif_entropy_irq_disable_all()` to temporary
+ * interrupt save-and-restore.
+ *
+ * @param entropy An entropy source handle.
+ * @param snapshot A snapshot to restore from.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_restore_all(
+    const dif_entropy_t *entropy, const dif_entropy_irq_snapshot_t *snapshot);
+
+/**
+ * Forces a particular alert, causing it to be emitted as if the hardware had
+ * done so.
+ *
+ * @param entropy An entropy source handle.
+ * @param alert An alert type.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_alert_force(const dif_entropy_t *entropy,
+                                             dif_entropy_alert_t alert);
+
+/**
+ * Performs an override read from the entropy pipeline.
```

Comment:
nit: "Performs" sounds like this is a one-time thing. The following comment is pretty clear but maybe this can be reworded.